lib/contentful.js

```diff
@@ -201,69 +201,90 @@ module.exports.renderMessageForPhoenixCampaign = function (phoenixCampaign, msgT
   return new Promise((resolve, reject) => {
     const campaignFieldNames = Object.keys(config.campaignFields);
     const campaignContent = {};
-    const result = {};
-
-    return exports.fetchCampaign(phoenixCampaign.id)
+    const campaignFields = {};
+    return exports.fetchCampaign(campaignId)
       .then((contentfulCampaign) => {
         campaignContent.overrides = contentfulCampaign.fields;
-
         return exports.fetchCampaign(defaultCampaignId);
       })
       .then((defaultContentfulCampaign) => {
         campaignContent.defaults = defaultContentfulCampaign.fields;
-        const promises = [];
-
         campaignFieldNames.forEach((key) => {
           const fieldName = config.campaignFields[key];
-          result[fieldName] = {};
+          campaignFields[fieldName] = {};
 
           if (campaignContent.overrides[fieldName]) {
-            result[fieldName].override = true;
-            result[fieldName].raw = campaignContent.overrides[fieldName];
+            campaignFields[fieldName].override = true;
+            campaignFields[fieldName].raw = campaignContent.overrides[fieldName];
           } else {
-            result[fieldName].override = false;
-            result[fieldName].raw = campaignContent.defaults[fieldName];
+            campaignFields[fieldName].override = false;
+            campaignFields[fieldName].raw = campaignContent.defaults[fieldName];
           }
-
-          promises.push(helpers.replacePhoenixCampaignVars(result[fieldName].raw, phoenixCampaign));
         });
-
-        return Promise.all(promises);
-      })
-      .then((renderedMessages) => {
-        // The order of renderedMessages corresponds to the sequence we defined campaignFieldNames.
-        renderedMessages.forEach((renderedMessage, index) => {
-          // Store the machine name used in code (e.g. 'ask_caption').
-          const key = campaignFieldNames[index];
-          // Find the Contentful field name that corresponds to the machine name.
-          const fieldName = config.campaignFields[key];
-
-          if (result[fieldName]) {
-            result[fieldName].rendered = renderedMessage;
-          }
-        });
-
-        return resolve(result);
+        return resolve(campaignFields);
       })
       .catch(err => reject(err));
   });
 };
 
-// Utility functions
-
-module.exports.getContentfulMessageTypes = function getCampaignIdFields() {
-  return Object.keys(config.campaignFields);
+/**
+ * Returns all rendered messages for given phoenixCampaign, using defaults when no override defined.
+ */
+module.exports.renderAllMessagesForPhoenixCampaign = function (phoenixCampaign) {
+  logger.debug(`renderAllMessagesForPhoenixCampaign:${phoenixCampaign.id}`);
+  return new Promise((resolve, reject) =>
+    // get all fields, overrides and default
+    exports.getAllFieldsForCampaign(phoenixCampaign.id)
+      .then((campaignFields) => { // eslint-disable-line arrow-body-style
```

Comment:
Why do we need to disable `arrow-body-style` here? would this be resolved by changing line 254 to `Promise.map(` instead of `return Promise.map(`?